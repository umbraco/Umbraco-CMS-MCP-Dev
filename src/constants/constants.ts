export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
