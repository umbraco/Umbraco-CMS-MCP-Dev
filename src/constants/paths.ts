import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');