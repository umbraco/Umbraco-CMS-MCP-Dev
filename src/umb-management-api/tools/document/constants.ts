export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}