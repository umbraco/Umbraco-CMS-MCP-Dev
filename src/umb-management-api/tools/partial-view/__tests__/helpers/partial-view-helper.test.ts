import { PartialViewHelper } from "./partial-view-helper.js";
import { PartialViewBuilder } from "./partial-view-builder.js";
import { PartialViewFolderBuilder } from "./partial-view-folder-builder.js";
import { BLANK_UUID } from "@/constants/constants.js";
import { jest } from "@jest/globals";

const TEST_PARTIAL_VIEW_NAME = "_TestHelperPartialView";
const TEST_FOLDER_NAME = "_TestHelperFolder";
const TEST_CONTENT = "@* Test helper content *@\n<p>Helper Test</p>";

describe("PartialViewHelper", () => {
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    await PartialViewHelper.cleanup(TEST_PARTIAL_VIEW_NAME + ".cshtml");
    await PartialViewHelper.cleanup(TEST_FOLDER_NAME);
  });

  it("should verify partial view exists after creation", async () => {
    // Arrange
    const builder = new PartialViewBuilder()
      .withName(TEST_PARTIAL_VIEW_NAME)
      .withContent(TEST_CONTENT);
    
    await builder.create();

    // Act
    const exists = await PartialViewHelper.verifyPartialView(builder.getPath());

    // Assert
    expect(exists).toBe(true);
  });

  it("should verify partial view folder exists after creation", async () => {
    // Arrange
    const folderBuilder = new PartialViewFolderBuilder()
      .withName(TEST_FOLDER_NAME);
    
    await folderBuilder.create();

    // Act
    const exists = await PartialViewHelper.verifyPartialViewFolder(folderBuilder.getPath());

    // Assert
    expect(exists).toBe(true);
  });

  it("should find partial view by name", async () => {
    // Arrange
    const builder = new PartialViewBuilder()
      .withName(TEST_PARTIAL_VIEW_NAME)
      .withContent(TEST_CONTENT);
    
    await builder.create();

    // Act
    const found = await PartialViewHelper.findPartialView(TEST_PARTIAL_VIEW_NAME + ".cshtml");

    // Assert
    expect(found).toBeDefined();
    expect(found.name).toBe(TEST_PARTIAL_VIEW_NAME + ".cshtml");
  });

  it("should find partial views with snapshot normalization", async () => {
    // Arrange
    const builder = new PartialViewBuilder()
      .withName(TEST_PARTIAL_VIEW_NAME)
      .withContent(TEST_CONTENT);
    
    await builder.create();

    // Act
    const foundItems = await PartialViewHelper.findPartialViews(TEST_PARTIAL_VIEW_NAME + ".cshtml");

    // Assert
    expect(foundItems).toMatchSnapshot();
  });

  it("should find partial view folders", async () => {
    // Arrange
    const folderBuilder = new PartialViewFolderBuilder()
      .withName(TEST_FOLDER_NAME);
    
    await folderBuilder.create();

    // Act
    const foundFolders = await PartialViewHelper.findPartialViewFolders(TEST_FOLDER_NAME);

    // Assert
    expect(foundFolders).toHaveLength(1);
    expect(foundFolders[0].name).toBe(TEST_FOLDER_NAME);
    expect(foundFolders[0].isFolder).toBe(true);
  });

  it("should get partial view with snapshot normalization", async () => {
    // Arrange
    const builder = new PartialViewBuilder()
      .withName(TEST_PARTIAL_VIEW_NAME)
      .withContent(TEST_CONTENT);
    
    await builder.create();

    // Act
    const partialView = await PartialViewHelper.getPartialView(builder.getPath());

    // Assert
    expect(partialView).toMatchSnapshot();
  });

  it("should get partial view folder with snapshot normalization", async () => {
    // Arrange
    const folderBuilder = new PartialViewFolderBuilder()
      .withName(TEST_FOLDER_NAME);
    
    await folderBuilder.create();

    // Act
    const folder = await PartialViewHelper.getPartialViewFolder(folderBuilder.getPath());

    // Assert
    expect(folder).toMatchSnapshot();
  });

  it("should cleanup created items", async () => {
    // Arrange
    const builder = new PartialViewBuilder()
      .withName(TEST_PARTIAL_VIEW_NAME)
      .withContent(TEST_CONTENT);
    
    await builder.create();
    const path = builder.getPath();

    // Verify it exists
    let exists = await PartialViewHelper.verifyPartialView(path);
    expect(exists).toBe(true);

    // Act
    await PartialViewHelper.cleanup(TEST_PARTIAL_VIEW_NAME + ".cshtml");

    // Assert
    exists = await PartialViewHelper.verifyPartialView(path);
    expect(exists).toBe(false);
  });
});