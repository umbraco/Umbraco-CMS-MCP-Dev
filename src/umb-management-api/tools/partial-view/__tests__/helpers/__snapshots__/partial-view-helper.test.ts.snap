// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartialViewHelper should find partial views with snapshot normalization 1`] = `
[
  {
    "hasChildren": false,
    "isFolder": false,
    "name": "_TestHelperPartialView.cshtml",
    "parent": null,
    "path": "/_TestHelperPartialView.cshtml",
  },
]
`;

exports[`PartialViewHelper should get partial view folder with snapshot normalization 1`] = `
{
  "name": "_TestHelperFolder",
  "parent": {
    "path": "/",
  },
  "path": "/_TestHelperFolder",
}
`;

exports[`PartialViewHelper should get partial view with snapshot normalization 1`] = `
{
  "content": "@* Test helper content *@
<p>Helper Test</p>",
  "name": "_TestHelperPartialView.cshtml",
  "parent": null,
  "path": "/_TestHelperPartialView.cshtml",
}
`;
